packages/runtime/runtime-utils/src/summarizerNode.ts

```diff
@@ -45,7 +45,7 @@ async function seqFromTree(
 }
 
 /** Path for nodes in a tree with escaped special characters */
-class EscapedPath {
+export class EscapedPath {
```

Comment:
nit: I see you exported these so you could use them in the extended class, but as far as minimizing leaking implementation goes, you may want to somehow filter what gets exported from the package.

So these shared things can be exported from a file: `summarizerNodeCore.ts` (or whatever you choose to name it)
Then the things in this file which were already exported can remain `summarizerNode.ts`
And the things in your new file can be exported `summarizerNodeWithGc.ts`
But the runtime-utils package will not export the core file in `index.ts`.

Does that make sense?

Reply:
Yes, I totally agree with this.
I actually wanted to do something similar. Move out the pieces that are not in the core SummarizerNode class to a separate file, including some helper functions. That would make the summarizerNode.ts file smaller and cleaner.
But I don't want to do this as part of this change since its not related. I would rather do that in a separate PR that might end up cleaning the summarizer node some more.

Reply:
Well the only thing is, technically this PR will make that subsequent change a "breaking" change, though we don't need to announce that since (hopefully) nobody will directly reference these internals.

Reply:
But I'm totally fine to do it in separate PR.  Even I can do it if you'd like to make sure we get it in same release.

Reply:
Yes to getting in the same release. Thanks, I would really appreciate if you could help with this :-).

Reply:
Opened a PR for this #4447